arangod/Cluster/AgencyCache.cpp

```diff
@@ -151,18 +173,75 @@ void AgencyCache::handleCallbacksNoLock(VPackSlice slice, std::unordered_set<uin
       }
     }
   }
+
+  std::unordered_set<std::string> dbs;
+  const char SLASH ('/');
+
   // Find keys, which are a prefix of a callback:
   for (auto const& k : keys) {
+
     auto it = _callbacks.lower_bound(k);
     while (it != _callbacks.end() && it->first.compare(0, k.size(), k) == 0) {
       if (uniq.emplace(it->second).second) {
         toCall.push_back(it->second);
       }
       ++it;
     }
+
+    if (k.size() > 8) {
+      std::string_view r(k.c_str() + 8, k.size() - 8);
+      auto rs = r.size();
+
+      if (rs > strlen(PLAN) && r.compare(0, strlen(PLAN), PLAN) == 0) {
```

Comment:
why don't to use `std::string_view` and avoid tons of `strlen` calls?

Reply:
have taken the suggestion for a general overhaul that also goes into devel